### ISSUE TITLE
TypeError: object OAuth2Token can't be used in 'await' expression < cli.update_token >

### ISSUE THREAD

**Issue by arainboldt** [alexgolec/tda-api#180]
## Please read the [bug submission guidelines](https://tda-api.readthedocs.io/en/latest/help.html) before submitting a bug.

Not following guidelines may result in your bug being ignored and/or closed. 

**Description of Bug**

Token update fails when using the client constructed via `client_from_access_functions` and making multiple calls via `get_quotes` on long lists of symbols. 

**Code to Reproduce**

```
import json
import asyncio


api_key = ''
token_path = ''

def get_token_from_json(path):

    def get_token():
        with open(path,'rb') as f:
            tokens = json.load(f)
        return tokens

    return get_token

def write_token_to_json(path):

    def write_token(tokens,refresh_token=None,*args,**kwargs):
        if refresh_token is not None:
            tokens['refresh_token'] = refresh_token
        with open(path,'w+') as f:
            json.dump(tokens,f)
        return tokens

    return write_token

async def process_calls(joblist):
    responses = []
    for job in joblist:
        resp = await job
        responses.append( resp )
    return responses

cli = tda.auth.client_from_access_functions(
                api_key=api_key,
                token_read_func=get_token_from_json(os.path.expanduser(token_path)),
                token_write_func=write_token_to_json(os.path.expanduser(token_path)),
                asyncio=True,
            )

jobs = [ cli.get_quotes(tda_opt_symbs[:150]), cli.get_quotes(tda_opt_symbs[150:])]

loop = asyncio.new_event_loop()

try:
    response = loop.run_until_complete( process_calls( jobs ) )
except Exception as e:
    import traceback
    import sys
    exc_info = sys.exc_info()
    print(e)
    traceback.print_exception(*exc_info)
finally:
    loop.close()
    
```

IMPORTANT: Remember to anonymize your code. Be sure to replace API keys/Client IDs with placeholders. Also, never, ever share the contents of your token file. 

**Expected Behavior**

Expecting the calls not to fail.

**Actual Behavior**

Calls fail.

**Error/Exception Log, If Applicable**

```
Traceback (most recent call last):
  File "/tdameritrade/api.py", line 145, in pipeline
    response = loop.run_until_complete( self.process_calls(joblist) )
  File "lib/python3.8/site-packages/nest_asyncio.py", line 70, in run_until_complete
    return f.result()
  File "lib/python3.8/asyncio/futures.py", line 178, in result
    raise self._exception
  File "lib/python3.8/asyncio/tasks.py", line 280, in __step
    result = coro.send(None)
  File "tdameritrade/api.py", line 136, in process_calls
    resp = await job
  File "lib/python3.8/site-packages/tda/client/asynchronous.py", line 16, in _get_request
    resp = await self.session.get(dest, params=params)
  File "lib/python3.8/site-packages/httpx/_client.py", line 1548, in get
    return await self.request(
  File "lib/python3.8/site-packages/authlib/integrations/httpx_client/oauth2_client.py", line 90, in request
    await self.ensure_active_token()
  File "lib/python3.8/site-packages/authlib/integrations/httpx_client/oauth2_client.py", line 104, in ensure_active_token
    await self.refresh_token(url, refresh_token=refresh_token)
  File "
lib/python3.8/site-packages/authlib/integrations/httpx_client/oauth2_client.py", line 149, in _refresh_token
    await self.update_token(self.token, refresh_token=refresh_token)
TypeError: object OAuth2Token can't be used in 'await' expression
```

**Comment by arainboldt** [comment]
Seems to be resolved by converting the token_write_func to an async function, e.g. 

```
def write_token_to_json(path):

    async def write_token(tokens,refresh_token=None,*args,**kwargs):
        if refresh_token is not None:
            tokens['refresh_token'] = refresh_token
        with open(path,'w+') as f:
            json.dump(tokens,f)
        return tokens

    return write_token
```

Still not sure why it only fails periodically.

**Comment by alexgolec** [comment]
Fixed by #246 and #247
